retrieval/scrape.go

```diff
@@ -751,8 +751,10 @@ loop:
 		}
 
 		ref, ok := sl.cache.getRef(yoloString(met))
+		var lset labels.Labels
+		p.Metric(&lset)
```

Comment:
You can get this with `sl.cache.lsets[ref].lset` rather than having to build it (cleanup required).